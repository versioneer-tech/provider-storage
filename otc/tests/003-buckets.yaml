---
apiVersion: pkg.internal/v1beta1
kind: Storage
metadata:
  name: s-jane
spec:
  bucketAccessRequests:
  - bucketName: b-john
    reason: Need WriteOnly access
  buckets: []
  principal: jane
status:
  conditions:
  - lastTransitionTime: "2024-01-01T00:00:00Z"
    message: 'Unready resources: access-key-jane, observe-jane-restricted-b-john,
      and user-jane'
    reason: Creating
    status: "False"
    type: Ready
---
apiVersion: kubernetes.crossplane.io/v1alpha2
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: access-key-jane
  labels:
    crossplane.io/composite: s-jane
  name: access-key-jane
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-jane
    uid: ""
spec:
  forProvider:
    manifest:
      apiVersion: identity.opentelekomcloud.crossplane.io/v1alpha1
      kind: CredentialV3
      metadata:
        name: jane-aksk
      spec:
        forProvider:
          type: AK
          userIdRef:
            name: jane
        providerConfigRef:
          name: provider-otc
        writeConnectionSecretToRef:
          name: jane
          namespace: default
  providerConfigRef:
    kind: ProviderConfig
    name: provider-kubernetes
---
apiVersion: kubernetes.crossplane.io/v1alpha2
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: observe-jane-restricted-b-john
  labels:
    crossplane.io/composite: s-jane
  name: observe-jane-restricted-b-john
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-jane
    uid: ""
spec:
  forProvider:
    manifest:
      apiVersion: s3.opentelekomcloud.crossplane.io/v1alpha1
      kind: BucketPolicy
      metadata:
        name: bucketpolicy-b-john
  managementPolicies:
  - Observe
  providerConfigRef:
    kind: ProviderConfig
    name: provider-kubernetes
---
apiVersion: kubernetes.crossplane.io/v1alpha2
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: user-jane
  labels:
    crossplane.io/composite: s-jane
  name: user-jane
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-jane
    uid: ""
spec:
  forProvider:
    manifest:
      apiVersion: identity.opentelekomcloud.crossplane.io/v1alpha1
      kind: UserV3
      metadata:
        name: jane
      spec:
        forProvider:
          enabled: true
          name: jane
        providerConfigRef:
          name: provider-otc
        writeConnectionSecretToRef:
          name: jane-user
          namespace: default
  providerConfigRef:
    kind: ProviderConfig
    name: provider-kubernetes
