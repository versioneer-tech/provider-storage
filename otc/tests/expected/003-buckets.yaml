---
apiVersion: pkg.internal/v1beta1
kind: Storage
metadata:
  name: s-jane
spec:
  bucketAccessRequests:
  - bucketName: b-john
    reason: Need WriteOnly access
  buckets: []
  principal: jane
status:
  conditions:
  - lastTransitionTime: "2024-01-01T00:00:00Z"
    message: 'Unready resources: observe-credential-jane, user-jane'
    reason: Creating
    status: "False"
    type: Ready
---
apiVersion: kubernetes.m.crossplane.io/v1alpha1
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: observe-credential-jane
  labels:
    crossplane.io/composite: s-jane
  name: observe-credential-jane
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-jane
    uid: ""
spec:
  forProvider:
    manifest:
      apiVersion: identity.opentelekomcloud.crossplane.io/v1alpha1
      kind: CredentialV3
      metadata:
        name: aksk-jane
  managementPolicies:
  - Observe
  providerConfigRef:
    kind: ProviderConfig
    name: provider-kubernetes
---
apiVersion: kubernetes.m.crossplane.io/v1alpha1
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: user-jane
  labels:
    crossplane.io/composite: s-jane
  name: user-jane
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-jane
    uid: ""
spec:
  forProvider:
    manifest:
      apiVersion: identity.opentelekomcloud.crossplane.io/v1alpha1
      kind: UserV3
      metadata:
        name: jane
      spec:
        forProvider:
          enabled: true
          name: jane
        providerConfigRef:
          name: provider-otc
  providerConfigRef:
    kind: ProviderConfig
    name: provider-kubernetes
